mlflow/projects/__init__.py

```diff
@@ -280,6 +301,23 @@ def _fetch_project(uri, force_tempdir, version=None, git_username=None, git_pass
     return res
 
 
+def _unzip_repo(zip_file, dst_dir):
+    import zipfile
+    with zipfile.ZipFile(zip_file) as zip_in:
+        zip_in.extractall(dst_dir)
+
+
+def _fetch_zip_repo(uri):
+    import requests
+    from io import BytesIO
+    response = requests.get(uri)
```

Comment:
Ideally we should use something like mlflow.data.download_uri() for this so that it can also support files in S3 and other cloud storage systems. We don't have a great API for it yet, so maybe we can commit this with just HTTP for now, but we should add that eventually and update this code to use it. You can leave a TODO here.

Reply:
Added a `TODO` that references a GitHub issue I created to track the progress of this change: https://github.com/mlflow/mlflow/issues/763